x-pack/legacy/plugins/maps/public/layers/util/data_request.js

```diff
@@ -37,5 +37,15 @@ export class DataRequest {
     return this._descriptor.dataId;
   }
 
+  getRequestToken() {
+    return this._descriptor.dataRequestToken;
+  }
+
+}
+
+export class DataRequestAbortError extends Error {
```

Comment:
Is it worth having this placeholder error without any specific message? Otherwise what's the gain of using this in `es_source` in addition to a normal Error?

Reply:
The gain is that it decouples the SearchSource abort error from VectorLayer.

VectorLayer just knows about DataRequestAbortError which is generic and can be thrown from any source while SearchSource  abort error is Elasticsearch source specific.